contrib/sarplus/python/tests/test_pyspark_sar.py

```diff
@@ -1,46 +1,52 @@
 import calendar
```

Comment:
quick question, are we using similar tests in recommenders SAR and in SAR+? ideally both algos will need to have the same results

Reply:
Could you tell me what's recommenders SAR?

Reply:
code: https://github.com/simonzhaoms/recommenders/blob/simonz/sarplus/spark3/recommenders/models/sar/sar_singlenode.py
tests: https://github.com/simonzhaoms/recommenders/blob/simonz/sarplus/spark3/tests/unit/recommenders/models/test_sar_singlenode.py

Reply:
I think, except the `topk` tests ([`test_get_item_based_topk`](https://github.com/simonzhaoms/recommenders/blob/b5d3b55355a88b8e7b847d92eb0f15d31a76b665/tests/unit/recommenders/models/test_sar_singlenode.py#L259), [`test_get_popularity_based_topk`](https://github.com/simonzhaoms/recommenders/blob/b5d3b55355a88b8e7b847d92eb0f15d31a76b665/tests/unit/recommenders/models/test_sar_singlenode.py#L308), [`test_get_normalized_scores`](https://github.com/simonzhaoms/recommenders/blob/b5d3b55355a88b8e7b847d92eb0f15d31a76b665/tests/unit/recommenders/models/test_sar_singlenode.py#L326)), they are similar.  But I am not sure whether they can get the same results.  May need to dig further.

Reply:
> But I am not sure whether they can get the same results. May need to dig further.

Both algos should get the same results, they are deterministic. In the tests we have similarity with low tolerance https://github.com/simonzhaoms/recommenders/blob/simonz/sarplus/spark3/tests/conftest.py#L94

Reply:
@simonzhaoms this is important, can you confirm that we are getting the same results in the tests?

Reply:
[The following tests](https://github.com/simonzhaoms/recommenders/blob/8c44d14e4402e0f06f4fa1838f94b9592b159848/contrib/sarplus/python/tests/test_pyspark_sar.py#L180-L386) in both SAR and SAR+ have the same results:
- `test_fit`
- `test_sar_item_similarity`
- `test_user_affinity`
- `test_userpred` of SAR+ is the same as `test_recommend_k_items` of SAR

But SAR has [several extra advanced tests](https://github.com/simonzhaoms/recommenders/blob/8c44d14e4402e0f06f4fa1838f94b9592b159848/tests/unit/recommenders/models/test_sar_singlenode.py#L259-L382) that may be newly added but not implemented in SAR+:
- `test_get_item_based_topk`
- `test_get_popularity_based_topk`
- `test_get_normalized_scores`